flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java

```diff
@@ -1359,28 +1359,28 @@ private void setMeasuredDimensionForFlex(@FlexDirection int flexDirection, int w
      * @param childLength   the length of a child view which is to be collected to the flex line
      * @param lp            the LayoutParams for the view being determined whether a new flex line
      *                      is needed
-     * @param isHorizontal  the direction of the layout
      * @return {@code true} if a wrap is required, {@code false} otherwise
      * @see #getFlexWrap()
      * @see #setFlexWrap(int)
      */
     private boolean isWrapRequired(int flexWrap, int mode, int maxSize,
-                                   int currentLength, int childLength, LayoutParams lp, boolean isHorizontal) {
+            int currentLength, int childLength, LayoutParams lp) {
         if (flexWrap == FLEX_WRAP_NOWRAP) {
             return false;
         }
         if (lp.wrapBefore) {
             return true;
         }
 
-        // the text wrap issue is come from here, because it didn't take margins into account
+        // the issue of text being cut in TextView with margin comes from here, because it didn't take margins into account
         // so i add the margin into the calculation to fix the bug
-        if (isHorizontal)
+        if (mFlexDirection == FLEX_DIRECTION_ROW || mFlexDirection == FLEX_DIRECTION_ROW_REVERSE) {
```

Comment:
Could you revert the isWrapRequired method?

As in my previous comment, you can add lp.leftMargin + lp.rightMargin (or lp.topMargin + lp.bottomMargi) to the fourth when isWrapRequired is called in measureHorizontal/measureVertical